universum/modules/structure_handler.py

```diff
@@ -136,24 +137,25 @@ def block(self, *, block_name: str, pass_errors: bool) -> Generator:
             self.close_block()
 
     def execute_one_step(self, configuration: Step,
-                         step_executor: Callable) -> Optional[str]:
+                         step_executor: Callable):
         # step_executor is [[Step], Step], but referring to Step creates circular dependency
         process = step_executor(configuration)
 
         error: Optional[str] = process.start()
         if error is not None:
-            return error
+            return process, error
```

Comment:
Instead of returning tuple, I suggest to perform some refactoring:
1. Declare interface for running step explicitly, as an abstract base class
2. Use type declaration of this interface for type checking with mypy
3. Include error status into this interface instead of returning it from the "start" and "finalize" functions separately
4. Pass the object into active_background_steps instead of passing separate methods

Reply:
As discussed in chat, will be done in a separate PR and then merged here.